### ISSUE TITLE
Enums raise exception when used in input object

### ISSUE THREAD

**Issue by grimmigerFuchs** [graphql-crystal/graphql#17]
Minimal example:

```crystal
require "kemal"
require "graphql"

@[GraphQL::Enum]
enum MyEnum
  Type1
  Type2
end

@[GraphQL::InputObject]
class MyInput
  include GraphQL::InputObjectType

  getter my_enum

  @[GraphQL::Field]
  def initialize(@my_enum : MyEnum)
  end
end

@[GraphQL::Object]
class Query
  include GraphQL::ObjectType
  include GraphQL::QueryType

  @[GraphQL::Field]
  def my_query(input : MyInput) : MyEnum
    input.my_enum
  end
end

SCHEMA = GraphQL::Schema.new(Query.new)

post "/graphql" do |env|
  env.response.content_type = "application/json"

  query = env.params.json["query"].as(String)
  variables = env.params.json["variables"]?.as(Hash(String, JSON::Any)?)
  operation_name = env.params.json["operationName"]?.as(String?)

  SCHEMA.execute(query, variables, operation_name)
end

Kemal.run

```

When calling `my_query` with `MyInput` the following exception gets raised:

```json
{
  "data": {},
  "errors": [
    {
      "message": "wrong type for argument input",
      "path": [
        "myQuery"
      ]
    }
  ]
}
```

When using the enum directly as an argument it works correctly.

I fixed this temporarily by adding a `GraphQL::Language::AEnum` case resulting to this diff:

```diff
diff --git a/src/graphql/input_object_type.cr b/src/graphql/input_object_type.cr
index b604636..07f9d2c 100644
--- a/src/graphql/input_object_type.cr
+++ b/src/graphql/input_object_type.cr
@@ -53,6 +53,8 @@ module GraphQL::InputObjectType
                     end.as({{inner_type.id}})
                   end.as(Array({{inner_type.id}}))
                 {% end %}
+                when ::GraphQL::Language::AEnum
+                  {{type}}.parse(arg_value.name)
                 when {{type.id}}
                   arg_value
                 else

```

Because of this the tests are not able to run successfully.

Edit: The fix doesn't work in other usecases neither.

**Comment by jgillich** [comment]
> Edit: The fix doesn't work in other usecases neither.

Can you elaborate?

**Comment by grimmigerFuchs** [comment]
I tried using it in a fork but it does not function as it's faulty in hindsight.

**Comment by jgillich** [comment]
Fixed by ddc724d87de979fb15df8aa0d5ea04b5997ea419
